Arriba/Arriba.Web/parts/Index.jsx

```diff
@@ -35,15 +91,89 @@ class Index extends React.Component {
             }
         );
     }
+    getCounts(then) {
+        // On query, ask for the count from every table.
+        this.timer = null;
+
+        if (!this.state.query) {
+            this.setState({ counts: undefined, loading: false });
+            return;
+        }
+
+        // Notify any listeners (such as the loading animation).
+        this.setState({ loading: true });
+
+        // Get the count of matches from each accessible table
+        xhr("allCount", { q: this.state.query })
+            .then(data => {
+                this.setState({ counts: data, loading: false }, then);
+
+                data.parsedQuery = data.parsedQuery.replace(/\[\*\]:/g, ""); // Other consumers want the [*] removed also.
+                this.mru.update(data.parsedQuery);
+            });
+    }
+
     render() {
         if (this.state.blockingErrorStatus != null) return <ErrorPage status={this.state.blockingErrorStatus} />;
 
-        const Page = window.location.pathname.startsWith("/Grid.html") ? Grid : Search;
-        return <div className="viewport">
+        const Page = this.state.mode ? Grid : Search;
```

Comment:
Hmm. If we can store a React class in state, maybe we can just store the class itself to use as "Page" in this.state.mode.

Reply:
I'm pretty sure we can store it. It depends on how much state you want to maintain when switching back and forth.